test/e2e/all_in_one_test.go

```diff
@@ -230,6 +236,115 @@ func TestDeployAllInOnePostgres(t *testing.T) {
 	verifyIngress(ctx, t, env)
 }
 
+func TestDeployAllInOnePostgresReplicas(t *testing.T) {
+	t.Log("configuring all-in-one-postgres.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageOverride); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithAddons(addons...)
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.Parse(clusterVersionStr)
+		require.NoError(t, err)
+		builder.WithKubernetesVersion(clusterVersion)
+	}
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, postgresPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+
+	t.Log("this deployment used a postgres backend, verifying that postgres migrations ran properly")
+	verifyPostgres(ctx, t, env)
+
+	t.Log("running ingress tests to verify all-in-one deployed ingress controller and proxy are functional")
+	verifyIngress(ctx, t, env)
+
+	forDeployment := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", deployment.Name),
+	}
+
+	podList, err := env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+	require.NoError(t, err)
+	require.Equal(t, 1, len(podList.Items))
+	initialPod := podList.Items[0]
+
+	scale := &autoscalingv1.Scale{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      deployment.Name,
+			Namespace: deployment.Namespace,
+		},
+		Spec: autoscalingv1.ScaleSpec{
+			Replicas: 2,
+		},
+	}
+	t.Log("adding a second replica to the Kong deployment")
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).UpdateScale(ctx,
+		deployment.Name, scale, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	require.Eventually(t, func() bool {
+		deployment, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Get(ctx, deployment.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		return deployment.Status.ReadyReplicas == *deployment.Spec.Replicas
+	}, kongComponentWait, time.Second)
+
+	podList, err = env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+	require.NoError(t, err)
+	var secondary corev1.Pod
+	for _, pod := range podList.Items {
+		if pod.Name != initialPod.Name {
+			secondary = pod
+			break
+		}
+	}
+
+	t.Log("confirming the second replica is not the leader and is not pushing configuration")
+	forwardCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	startPortForwarder(forwardCtx, t, env, secondary, "9777", "cmetrics")
```

Comment:
is there a reason we're relying on a port-forward here? commonly in our tests we create a `LoadBalancer` service on kind clusters for this kind of thing and we already have test tooling relevant to this :thinking:

Reply:
LoadBalancer distributes requests at random. We wouldn't know if we're hitting the leader or the the follower node for any individual request.

Ideally we'd use Pod DNS instead, but we're not in the cluster, so :shrug: